src/ray/common/status.h

```diff
@@ -133,6 +134,7 @@ class RAY_EXPORT Status {
   ~Status() { delete state_; }
 
   Status(StatusCode code, const std::string &msg, int rpc_code = -1);
+  Status(StatusCode code, const std::string &msg, SourceLocation loc, int rpc_code = -1);
```

Comment:
adding a constructor that takes an rvalue string too might be a small thing that's worth it or just forwarding string and defining constructor in header?

Reply:
I think the correct way is to take value type, with no lvaule/rvalue overload.
Checkout "modern effective C++" item-41 for details.

I could followup with another PR later.